django_prbac/models.py

```diff
@@ -144,8 +144,12 @@ def get_privileges(self, assignment):
                 return self._granted_privileges
             except AttributeError:
                 pass
-        return [membership.instantiated_to_role(assignment)
-                for membership in self.memberships_granted.all()]
+        try:
+            return [membership.instantiated_to_role(assignment)
+                    for membership in self.memberships_granted.all()]
+        except ValueError:
+            # Django 4 raises ValueError if fk relationship is accessed prior to save
+            return []
```

Comment:
Given how many different things could raise `ValueError`, and we really want this catch to be as narrow as possible, do you think it's worth minimizing the code inside the try block? Would this work?
```suggestion
        try:
            memberships = self.memberships_granted.all()
        except ValueError:
            # Django 4 raises ValueError if fk relationship is accessed prior to save
            return []
        return [m.instantiated_to_role(assignment) for m in memberships]
```